lib/jsonld-signatures.js

```diff
@@ -121,3 +135,44 @@ api.purposes = require('./purposes').purposes;
 
 // expose document loader helpers
 Object.assign(api, require('./documentLoader'));
+
+/**
+ * Ensures the document to be signed contains the required signature suite
+ * specific `@context`, by either adding it (if `addSuiteContext` is true),
+ * or throwing an error if it's missing.
+ *
+ * @param {object} options - Options hashmap.
+ * @param {object} options.document - A JSON-LD document.
+ * @param {string} options.contextUrl - A context url.
+ * @param {boolean} options.addSuiteContext - Add suite context?
+ */
+function _ensureSuiteContext({document, contextUrl, addSuiteContext}) {
+  if(!_includesContext({document, contextUrl})) {
+    if(addSuiteContext) {
+      // Enforce the suite's context by adding it to the document
+      document['@context'] = [...document['@context'] || [], contextUrl];
+    } else {
+      throw new TypeError(
+        `The document to be signed must contain this suite's @context, ` +
+        `"${contextUrl}".`
+      );
+    }
+  }
+  // document does contain the context, do nothing
```

Comment:
Preferred pattern for reducing nesting and getting error conditions/early returns out of the way first:

```suggestion
  if(_includesContext({document, contextUrl})) {
    // document already includes the required context
    return;
  }

  if(!addSuiteContext) {
    throw new TypeError(
      `The document to be signed must contain this suite's @context, ` +
      `"${contextUrl}".`);
  }

  // enforce the suite's context by adding it to the document
  document['@context'] = [...document['@context'] || [], contextUrl];
```

Reply:
Thanks, fixing!

Reply:
Fixed.